third-party/cmake/BuildLuajit.cmake

```diff
@@ -1,25 +1,90 @@
+		  AND NOT _luajit_INSTALL_COMMAND)
+    message(FATAL_ERROR "Must pass at least one of CONFIGURE_COMMAND, BUILD_COMMAND, INSTALL_COMMAND")
+  endif()
+  if(NOT _luajit_TARGET)
+    set(_luajit_TARGET "luajit")
+  endif()
+
+  ExternalProject_Add(${_luajit_TARGET}
+    PREFIX ${DEPS_BUILD_DIR}
+    URL ${LUAJIT_URL}
+    DOWNLOAD_DIR ${DEPS_DOWNLOAD_DIR}/luajit
+    DOWNLOAD_COMMAND ${CMAKE_COMMAND}
+      -DPREFIX=${DEPS_BUILD_DIR}
+      -DDOWNLOAD_DIR=${DEPS_DOWNLOAD_DIR}/luajit
+      -DURL=${LUAJIT_URL}
+      -DEXPECTED_SHA256=${LUAJIT_SHA256}
+      -DTARGET=${_luajit_TARGET}
+      -P ${CMAKE_CURRENT_SOURCE_DIR}/cmake/DownloadAndExtractFile.cmake
+    CONFIGURE_COMMAND "${_luajit_CONFIGURE_COMMAND}"
+    BUILD_IN_SOURCE 1
+    BUILD_COMMAND "${_luajit_BUILD_COMMAND}"
+    INSTALL_COMMAND "${_luajit_INSTALL_COMMAND}")
+endfunction()
+
+set(INSTALLCMD_UNIX ${MAKE_PRG} CC=${DEPS_C_COMPILER}
+                                PREFIX=${DEPS_INSTALL_DIR}
+                                CFLAGS=-fPIC
+                                CFLAGS+=-DLUAJIT_DISABLE_JIT
+                                CFLAGS+=-DLUA_USE_APICHECK
+                                CFLAGS+=-DLUA_USE_ASSERT
+                                CCDEBUG+=-g
+                                BUILDMODE=static
+                                install
+    )
+
+if(UNIX)
+  BuildLuaJit(INSTALL_COMMAND ${INSTALLCMD_UNIX})
+
+elseif(MINGW AND CMAKE_CROSSCOMPILING)
+
+  # Build luajit for the host
+  BuildLuaJit(TARGET luajit_host
+	  CONFIGURE_COMMAND ""
+	  BUILD_COMMAND ""
+          INSTALL_COMMAND ${INSTALLCMD_UNIX} PREFIX=${HOSTDEPS_INSTALL_DIR} CC=${HOST_C_COMPILER})
+
+  # Build luajit for the target
+  BuildLuaJit(
+          # Similar to Unix + cross - fPIC
+	  INSTALL_COMMAND ${MAKE_PRG} PREFIX=${DEPS_INSTALL_DIR}
+          BUILDMODE=static install
+          TARGET_SYS=${CMAKE_SYSTEM_NAME}
+          CROSS=${CROSS_TARGET}-
+          HOST_CC=${HOST_C_COMPILER} HOST_CFLAGS=${HOST_C_FLAGS}
+          HOST_LDFLAGS=${HOST_EXE_LINKER_FLAGS}
+          FILE_T=luajit.exe
+          INSTALL_TSYMNAME=luajit.exe
+	  )
+
+elseif(WIN32 AND MSVC)
+
+  BuildLuaJit(
+          BUILD_COMMAND ${CMAKE_COMMAND} -E chdir ${DEPS_BUILD_DIR}/src/luajit/src ${DEPS_BUILD_DIR}/src/luajit/src/msvcbuild.bat static
+          INSTALL_COMMAND ${CMAKE_COMMAND} -E make_directory ${DEPS_INSTALL_DIR}/bin
+	    COMMAND ${CMAKE_COMMAND} -E copy ${DEPS_BUILD_DIR}/src/luajit/src/luajit.exe ${DEPS_INSTALL_DIR}/bin
+	    COMMAND ${CMAKE_COMMAND} -E copy ${DEPS_BUILD_DIR}/src/luajit/src/lua51.lib ${DEPS_INSTALL_DIR}/bin
+	    COMMAND ${CMAKE_COMMAND} -E make_directory ${DEPS_INSTALL_DIR}/lib
+	    COMMAND ${CMAKE_COMMAND} -E copy ${DEPS_BUILD_DIR}/src/luajit/src/lua51.lib ${DEPS_INSTALL_DIR}/lib
+	    COMMAND ${CMAKE_COMMAND} -E make_directory ${DEPS_INSTALL_DIR}/include/luajit-2.0
+	    # TODO: Copy only the necessary headers, but cmake does not accept wildcards
```

Comment:
You could create a separate CMake script (like `InstallLuajitHeaders.cmake`) and use `file(GLOB ..` in there.